<launch>
  <arg name="INPUT_IMAGE" />
  <arg name="INPUT_IMAGE_HEIGHT_SCALE" />
  <arg name="INPUT_IMAGE_WIDTH_SCALE" />
    <!--
      we need to scale interaction from image_view2
      1. screenractangle
      2. foreground
      3. background
      4. line
      5. poly
    -->
    <!-- 1. screenractangle -->
    <node pkg="image_view2" type="scale_interaction.py" name="scale_screenractangle"
          output="log">
      <remap from="~input" to="$(arg INPUT_IMAGE)/screenrectangle" />
      <remap from="~output" to="$(arg INPUT_IMAGE)/screenrectangle/scaled" />
      <remap from="~input/width_scale" to="$(arg INPUT_IMAGE_WIDTH_SCALE)" />
      <remap from="~input/height_scale" to="$(arg INPUT_IMAGE_HEIGHT_SCALE)" />
      <rosparam>
        inverse: true
      </rosparam>
    </node>
    <!-- 2. foreground -->
    <node pkg="image_view2" type="scale_interaction.py" name="scale_foreground"
          output="log">
      <remap from="~input" to="$(arg INPUT_IMAGE)/foreground_rect" />
      <remap from="~output" to="$(arg INPUT_IMAGE)/foreground_rect/scaled" />
      <remap from="~input/width_scale" to="$(arg INPUT_IMAGE_WIDTH_SCALE)" />
      <remap from="~input/height_scale" to="$(arg INPUT_IMAGE_HEIGHT_SCALE)" />
      <rosparam>
        inverse: true
      </rosparam>
    </node>
    <!-- 3. background -->
    <node pkg="image_view2" type="scale_interaction.py" name="scale_background"
          output="log">
      <remap from="~input" to="$(arg INPUT_IMAGE)/background_rect" />
      <remap from="~output" to="$(arg INPUT_IMAGE)/background_rect/scaled" />
      <remap from="~input/width_scale" to="$(arg INPUT_IMAGE_WIDTH_SCALE)" />
      <remap from="~input/height_scale" to="$(arg INPUT_IMAGE_HEIGHT_SCALE)" />
      <rosparam>
        inverse: true
      </rosparam>
    </node>
    <!-- 4. line -->
    <node pkg="image_view2" type="scale_interaction.py" name="scale_line"
          output="log">
      <remap from="~input" to="$(arg INPUT_IMAGE)/line" />
      <remap from="~output" to="$(arg INPUT_IMAGE)/line/scaled" />
      <remap from="~input/width_scale" to="$(arg INPUT_IMAGE_WIDTH_SCALE)" />
      <remap from="~input/height_scale" to="$(arg INPUT_IMAGE_HEIGHT_SCALE)" />
      <rosparam>
        inverse: true
      </rosparam>
    </node>
    <!-- 5. line -->
    <node pkg="image_view2" type="scale_interaction.py" name="scale_poly"
          output="log">
      <remap from="~input" to="$(arg INPUT_IMAGE)/poly" />
      <remap from="~output" to="$(arg INPUT_IMAGE)/poly/scaled" />
      <remap from="~input/width_scale" to="$(arg INPUT_IMAGE_WIDTH_SCALE)" />
      <remap from="~input/height_scale" to="$(arg INPUT_IMAGE_HEIGHT_SCALE)" />
      <rosparam>
        inverse: true
      </rosparam>
    </node>

</launch>
